tools/edit_instance_template/edit_instance_template.go

```diff
@@ -16,152 +17,276 @@ import (
 )
 
 var (
-	project    = flag.String("project", "", "GCP project that hosts the template")
-	template   = flag.String("template", "", "The name of the base template")
-	suffix     = flag.String("suffix", "ssd-network", "Suffix to append to instance template name")
-	size       = flag.Int("size", 5, "The initial number of instances in the group")
-	region     = flag.String("region", "us-west1", "The region of the subnetwork")
+	project       = flag.String("project", "", "GCP project that hosts the template")
+	region        = flag.String("region", "us-west1", "The region of the subnetwork")
+	templateRegex = flag.String("template_regex", "", "The regex to use when matching instance templates")
```

Comment:
It would be good to have a safer default for the template regex. I don't like the idea of modifying *all* MIGs globally by default, even with the dry-run option. 
What do you think of making the following changes:
1) add --gke_cluster flag and ignore any templates/MIGs that do not start with gke-{cluster}
2) change default for template_regex to .*executor.*
3) only modify instance templates used by MIGs in --region
the last one is probably the most annoying because you can't tell in which region a template is being used w/o looking at MIGs first. 

This might be a little paranoid, but I've read too many postmortems about people blowing things up with scripts like this :)

Reply:
I added a cluster flag and a pool flag, and I filter igms to only those in the specified region.